### PR TITLE
User can sign in

### PR DESCRIPTION
**Pull request by raomanasa**
[PT-LINK](https://www.pivotaltracker.com/story/show/170152677)
- adds login functionality
- adds sad path (when there is wrong email/password)

### CODE REVIEW

File: features/step_definitions/background_step.rb [hunk 1]
```diff
@@ -4,4 +4,9 @@
     product_attributes.except!("category")
     create(:product, product_attributes.merge(category: category))
   end
+end
+  Given("the following user exist:") do |table|
```
Comment (oliverochman) [inline features/step_definitions/background_step.rb]: fix indentation

File: features/step_definitions/basic_steps.rb [hunk 2]
```diff
@@ -12,4 +12,8 @@
 
 When("I fill in {string} with {string}") do |input, content|
   fill_in input, with: content
+end
+
+Given("I am on the login page") do
+ visit new_user_session_path
```
Comment (oliverochman) [inline features/step_definitions/basic_steps.rb]: keep indentation consistent

File: features/user_can_login.feature [hunk 3]
```diff
@@ -0,0 +1,31 @@
+Feature: User can log in
+
+As a user
+In order to access my account
+I would like to be able log in
+
+Background:
```
Comment (oliverochman) [inline features/user_can_login.feature]: fix indentation

File: features/user_can_login.feature [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        Given I am on the login page
```
Comment (oliverochman) [inline features/user_can_login.feature]: I need you to click your way to the login page, so we know that the link works

File: features/user_can_login.feature [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        And the following user exist:
```
Comment (oliverochman) [inline features/user_can_login.feature]: Populating the DB is the first thing we want to do in a backgroud

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: features/step_definitions/background_step.rb (hunk 1 above)
Comment (raomanasa) [reply features/step_definitions/background_step.rb]: ok.sure

File: features/step_definitions/basic_steps.rb (hunk 2 above)
Comment (raomanasa) [reply features/step_definitions/basic_steps.rb]: ok

File: features/user_can_login.feature (hunk 3 above)
Comment (raomanasa) [reply features/user_can_login.feature]: ok

File: features/user_can_login.feature (hunk 4 above)
Comment (raomanasa) [reply features/user_can_login.feature]: ok

File: features/user_can_login.feature (hunk 5 above)
Comment (raomanasa) [reply features/user_can_login.feature]: ok

**Review by oliverochman — APPROVED** [review]

**Pull request merged**